doc/acl/SONiC_ACL_based_metering_hld.md

```diff
@@ -0,0 +1,430 @@
+```
+Alternatively, user can use "config load" command to configure the policer action or use sonic-cli interface.
+
+```c
+{
+   	"POLICER": {
+		"policer1": {
+		    "meter_type": "bytes",
+		    "mode": "sr_tcm",
+		    "cir": "12500",
+		    "cbs": "12500",
+		    "pir": "17500",
+		    "pbs": "17500",
+		    "color": "aware",
+		    "red_packet_action": "drop",
+		    "yellow_packet_action": "drop",
+		    "green_packet_action": "forward"
+        },
+        "policer2": {
+		    "meter_type": "bytes",
+		    "mode": "sr_tcm",
+		    "cir": "12500",
+		    "cbs": "12500",
+		    "pir": "17500",
+		    "pbs": "17500",
+		    "color": "aware",
+		    "red_packet_action": "drop",
+		    "yellow_packet_action": "drop",
+		    "green_packet_action": "forward"
+        }
+	},
+    "ACL_TABLE": {
+        "VRF_ACL_REDIRECT_V4": {
+            "policy_desc": "Redirect traffic to nexthop in different vrfs",
+            "type": "L3",
+            "ports": ["Ethernet1"]
+        },
+        "VRF_ACL_REDIRECT_V6": {
+            "policy_desc": "Redirect traffic to nexthop in different vrfs",
+            "type": "L3V6",
+            "ports": ["Ethernet1"]
+        }
+    },
+    "ACL_RULE": {
+        "VRF_ACL_REDIRECT_V4|rule1": {
+            "priority": "55",
+            "SRC_IP": "30.0.0.1",
+            "policer_action":"policer1"
+        },
+        "VRF_ACL_REDIRECT_V6|rule1": {
+            "priority": "55",
+            "SRC_IPV6": "2000:0030::1",
+            "policer_action":"policer2"
+        }
+    }
+}
+
+```
+
+#### Show command
+
+acl-loader script will be updated to display policer action in show output. 
+
+```c
+admin@sonic:~$ show acl rule
+Table     Rule          Priority    Action                     Match
+--------  ------------  ----------  -------------------------  ----------------------------
+EVERFLOW  RULE_3        9997        MIRROR INGRESS: everflow0  SRC_IP: 20.0.0.2/32
+                                    POLICER: policer0
+EVERFLOW  RULE_4        9996        MIRROR EGRESS : everflow1  L4_SRC_PORT: 4621
+                                    POLICER: policer1
+DATAACL   RULE_5        9995        REDIRECT: Ethernet8        IP_PROTOCOL: 126
+                                    POLICER: policer2
+DATAACL   RULE_6        9994        FORWARD                    L4_SRC_PORT: 179
+                                    POLICER: policer3
+```
+```
+admin@sonic:~$ show acl table
+Name                 Type    Binding     Description          Stage    Action-list    Status
+-------------------  ------  ----------  -------------------  -------  ------------   -------
+SAMPLE_ACL_TABLE_V6  L3V6    Ethernet0   SAMPLE_ACL_TABLE_V6  ingress  packet-action  Active
+                             Ethernet1                                 policer
+                             Ethernet2                                                                  
+                             Ethernet3 
+                             Ethernet4
+
+```
```

Comment:
Could you include CRM for the policer resources.

Reply:
Based on vendor SAI, policer resources can be per stage.  And there is NO existing attribute or API to query the max and current object in SAI per STAGE. Will check with SAI community and update the HLD accordingly.